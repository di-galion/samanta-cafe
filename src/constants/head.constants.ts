export const ORDER_ARRAY_OF_HEAD = ["Имя", "Телефон", "Улица", "Дом", "Подьезд", "Этаж","Комната", "Ключ", "Комментарий"]

export const PRODUCT_ARRAY_OF_HEAD = ["Название", "Описание", "Цена"]

export const CATEGORY_ARRAY_OF_HEAD = ["Название", "Slug"]