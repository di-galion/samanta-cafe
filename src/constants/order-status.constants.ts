export const NO_FILTER = 'No filtration'