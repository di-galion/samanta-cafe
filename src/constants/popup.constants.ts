export const POPUP_ID = "popup"
export const POPUP_ITEM_TEXT  = "Do you want to save edit?"