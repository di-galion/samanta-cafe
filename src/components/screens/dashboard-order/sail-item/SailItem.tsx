"use client"

import {ChangeEvent, FC, useEffect, useRef, useState} from "react";
import styles from "./styles.module.scss";
import {RxCross1} from "react-icons/rx";
import {useMutation, useQueryClient} from "@tanstack/react-query";
import {SailService} from "@/services/sail/sail.service";
import {BiEditAlt} from "react-icons/bi";
import {GiConfirmed} from "react-icons/gi";
import Button from "@/components/ui/button/Button";
import {ISail} from "@/types/sail.types";
import cn from "classnames";
import {DELETE_SAIL, GET_ALL_ORDERS, UPDATE_SAIL} from "@/constants/query-keys.constants";
import {useUpdateSailMutation} from "@/components/screens/dashboard-order/sail-item/useUpdateSailMutation.hook";
import {useDeleteSailMutation} from "@/components/screens/dashboard-order/sail-item/useDeleteSailMutation.hook";          


const SailItem: FC<{sail: ISail}> = ({sail}) => {
    const [isEditing, setIsEditing] = useState(false)
    const [editItem, setEditItem] = useState({})
    // To prevent use Effect during initial render
    const isMounted = useRef(false);

    const mutateUpdate = useUpdateSailMutation(sail, editItem)
    const mutateDelete = useDeleteSailMutation(sail)

    useEffect(() => {
        if(isMounted.current) mutateUpdate()
         else isMounted.current = true
    }, [editItem])


    const onSubmit = () => {
        const form = new FormData(document.getElementById(`form-sail ${sail.id}`))
        if (form.get(sail.name) === null) return
        setEditItem({
            name: form.get(sail.name),
            price: +form.get(sail.price),
            quantity: +form.get(sail.quantity)
        })
    }

    return (
        <div className={styles.sail}>
            {/*<Image />*/}
            <form
                data-testid={"sail-item-test"}
                id={`form-sail ${sail.id}`}
                className={styles.info_section}
            >
                <div className={styles.options_wrapper}>
                    {!isEditing ?
                        <BiEditAlt
                            data-testid={"button-edit-test"}
                            size={23}
                            onClick={() => setIsEditing(!isEditing)}
                        /> :
                        <button
                            data-testid={"button-confirm-test"}
                            type="submit"
                            onClick={() => {
                                setIsEditing(false)
                                onSubmit()
                            }}
                        >
                            <GiConfirmed size={23} />
                        </button>
                    }
                    <RxCross1
                        data-testid={"button-delete-test"}
                        size={20}
                        className={styles.delete}
                        onClick={() => mutateDelete()}
                    />
                </div>

                <div className={styles.input_wrapper}>
                    {[
                        {name: sail.name, label: "Name"},
                        {name: sail.price, label: "Price"},
                        {name: sail.quantity, label: "Quantity"}].map((item) => {
                        return <SailInput key={item.label} name={item.name} label={item.label} isEditing={isEditing}/>
                    })}
                </div>
            </form>
        </div>
    )
}

export default SailItem

export interface ISailInput {
    name: string | number,
    label: string | number,
    isEditing: boolean
}
const SailInput: FC<ISailInput>= ({ name, label, isEditing}) => {
    return (
        <>
            <label htmlFor={String(name)}>{label}:</label>
            <input
                className={cn(styles.input, isEditing ? styles.active_input : "")}
                disabled={!isEditing} name={name.toString()} defaultValue={name}
            />
        </>
    )
}