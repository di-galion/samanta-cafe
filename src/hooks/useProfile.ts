import {useQuery} from "@tanstack/react-query";
import {CategoryService} from "@/services/category/category.service";
import {AuthService} from "@/services/auth/auth.service";
import {useActions} from "@/hooks/useActions";
import {useAuth} from "@/hooks/useAuth";
import {UserService} from "@/services/user/user.service";
import {is} from "immutable";
import {GET_PROFILE} from "@/constants/query-keys.constants";

export const useProfile = () => {
    const {user} = useAuth()

    // console.log(user)

    const {data} = useQuery({
        queryKey: [GET_PROFILE],
        queryFn: () => UserService.getProfile(),
        select: ({data}) => data,
        enabled: !!user
    })

    // console.log(data)

    return {profile: data || {}}
}