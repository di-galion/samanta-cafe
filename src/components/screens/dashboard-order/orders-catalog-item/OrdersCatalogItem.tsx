"use client"
import {ChangeEvent, FC, useEffect, useRef, useState} from "react";
import { IOrderResponse} from "@/types/order.types";
import styles from './styles.module.scss'
import Select from "@/components/ui/select/Select";
import {EnumOrderStatus} from "@/components/screens/dashboard-order/orders-catalog-item/types/types";
import {useMutation, useQueryClient} from "@tanstack/react-query";
import {OrderService} from "@/services/order/order.service";
import FormData from "form-data";
import {ORDER_ARRAY_OF_HEAD} from "@/constants/head.constants";
import {getOrderHeadValues} from "@/utils/get-head-values.utils";
import SailItem from "@/components/screens/dashboard-order/sail-item/SailItem";
import Button from "@/components/ui/button/Button";
import ProductList from "@/components/screens/dashboard-order/product-list/ProductList";
import {ISail, ISailCreate} from "@/types/sail.types";
import {SailService} from "@/services/sail/sail.service";
import OptionsWrapper from "@/components/ui/admin/options-wrapper/OptionsWrapper";
import InputsSectionWrapper from "@/components/ui/admin/inputs-section-wrapper/InputsSectionWrapper";
import EditFormWrapper from "@/components/ui/admin/edit-form-wrapper/EditFormWrapper";
import Popup from "@/components/ui/popup/Popup";
import {POPUP_ITEM_TEXT} from "@/constants/popup.constants";
import {CREATE_SAIL, GET_ALL_ORDERS, UPDATE_ORDER} from "@/constants/query-keys.constants";
import {IProduct} from "@/types/product.types";
import {useUpdateOrderMutationHook} from "@/components/screens/dashboard-order/orders-catalog-item/useUpdateOrderMutation.hook";
import {
    useCreateSailMutation
} from "@/components/screens/dashboard-order/orders-catalog-item/useCreateSailMutation.hook";



const OrdersCatalogItem: FC<{order: IOrderResponse}> = ({order}) => {
    const [isEditing, setIsEditing] = useState(false)
    const [editItem, setEditItem] = useState({})
    const [isProductListOpen, setIsProductListOpen] = useState(false)
    const [newSail, setNewSail] = useState<ISailCreate>()
    const [isPopupOpen, setIsPopupOpen] = useState<boolean>(false)
    // Tp prevent use Effect during initial render
    const isMountedOrder = useRef(false);
    const isMountedSail = useRef(false);

    const form = useRef(null);

    const arrayOfValue = getOrderHeadValues(order)

    const mutate = useUpdateOrderMutationHook(order, editItem)
    const mutateSailCreate = useCreateSailMutation(order, newSail)
    useEffect(() => {
        if (isMountedOrder.current) mutate()
        else isMountedOrder.current = true
    }, [editItem])
    useEffect(() => {
        if (isMountedSail.current) mutateSailCreate()
        else isMountedSail.current = true

    }, [newSail])

    const onSubmit = (formElement: HTMLFormElement | null) => {

        if (!formElement) return;
        const form = new FormData(formElement as HTMLFormElement)

        if (form.get(ORDER_ARRAY_OF_HEAD[0]) === null) return
        setEditItem({
            name: form.get(ORDER_ARRAY_OF_HEAD[0]),
            tel: form.get(ORDER_ARRAY_OF_HEAD[1]),
            street: form.get(ORDER_ARRAY_OF_HEAD[2]),
            house: form.get(ORDER_ARRAY_OF_HEAD[3]),
            entrance: form.get(ORDER_ARRAY_OF_HEAD[4]),
            floor: form.get(ORDER_ARRAY_OF_HEAD[5]),
            room: form.get(ORDER_ARRAY_OF_HEAD[6]),
            key: form.get(ORDER_ARRAY_OF_HEAD[7]),
        })
        // if (Object.keys(editItem).length) mutate()
    }

    return (
        <EditFormWrapper >
            <form
                data-testid={"order-item-test"}
                ref={form}
                onSubmit={(e: ChangeEvent<HTMLFormElement>) => onSubmit(e)}
            >
                <OptionsWrapper
                    setIsEditing={setIsEditing}
                    isEditing={isEditing}
                    onConfirmClick={setIsPopupOpen}
                >
                    <Select
                        testId={"change-status-test"}
                        onClick={(value) => {
                            setEditItem({status: value})
                        }}
                        currentValue={order.status}
                    >
                        {Object.entries(EnumOrderStatus).map((value) => {
                            return <li data-testid={`select-item-${value[1]}-test`} key={value[1]}>{value[1]}</li>})}
                    </Select>
                </OptionsWrapper>

                <InputsSectionWrapper
                    isEditing={isEditing}
                    arrayOfValue={arrayOfValue}
                    arrayOfHead={ORDER_ARRAY_OF_HEAD}
                />

                <div className={styles.sails}>
                    {order.sails.map((sail) => {
                        return (<SailItem key={sail.id} sail={sail} />)
                    })}
                </div>

                {isProductListOpen &&
                    <ProductList
                        orderId={order.id}
                        setNewSailOnClick={(sail: ISailCreate) => {
                            setNewSail(sail)
                            if (newSail !== undefined) {
                                setIsProductListOpen(false)
                            }
                        }}
                    />
                }

                <div className={styles.button_add_sails}>
                    <Button
                        type={"button"}
                        testId={"button-add-sails-test"}
                        onClick={() => setIsProductListOpen(!isProductListOpen)}
                    >
                        Add sails
                    </Button>
                </div>

                <div className={styles.total}>{order.total}</div>

                {isPopupOpen &&
                    <Popup
                        content={POPUP_ITEM_TEXT}
                        onClickAgreeHandler={(e: ChangeEvent<HTMLFormElement>) => {
                            setIsEditing(false)
                            onSubmit(form.current)
                        }}
                        onClickDisagreeHandler={() => {
                            setIsEditing(false)
                            if (form.current) form.current.reset()
                        }}
                        setIsOpen={setIsPopupOpen}
                    />
                }
            </form>
        </EditFormWrapper>

    )
}
export default OrdersCatalogItem
