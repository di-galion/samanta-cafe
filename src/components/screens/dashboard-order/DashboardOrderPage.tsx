"use client"

import Heading from "@/components/ui/heading/Heading";
import OrdersCatalog from "@/components/screens/dashboard-order/orders-catalog/OrdersCatalog";
import {useProfile} from "@/hooks/useProfile";
import {useRouter} from "next/navigation";
import OnlyForAdminText from "@/components/ui/admin/only-for-admin-text/OnlyForAdminText";
import PageContainer from "@/components/ui/page-container/PageContainer";

const DashboardOrderPage = () => {
    const {profile} = useProfile()
    const {replace} = useRouter()
    if(!profile?.isAdmin) {
        replace("/")
        return <></>
    }
    return (
        <PageContainer>
            <Heading >Orders</Heading>
            <OrdersCatalog />
        </PageContainer>
    )
}

export default DashboardOrderPage