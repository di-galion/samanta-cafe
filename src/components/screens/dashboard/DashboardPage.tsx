"use client"

import styles from "./styles.module.scss"
import Button from "@/components/ui/button/Button";
import Link from "next/link";
import {useProfile} from "@/hooks/useProfile";
import {useRouter} from "next/navigation";
const DashboardPage = () => {
    const {profile} = useProfile()
    const {replace} = useRouter()
    if(!profile?.isAdmin) {
        replace("/")
        return <></>
    }

    return (
        <div className={styles.dashboard_container}>
                <div className={styles.header}>
                    <Button >
                        <Link href={"/dashboard/product"} >Product panel</Link>
                    </Button>
                    <Button >
                        <Link href={"/dashboard/orders"} >Orders panel</Link>
                    </Button>
                    <Button >
                        <Link href={"/dashboard/category"} >Category panel</Link>
                    </Button>
                </div>
        </div>
    )
}

export default DashboardPage