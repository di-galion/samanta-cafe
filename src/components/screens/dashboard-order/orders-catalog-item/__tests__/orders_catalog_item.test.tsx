import {useProfile} from "@/hooks/useProfile";
import {renderWithProviders} from "@/__tests__/helpers/renderTestApp";
import DashboardPage from "@/components/screens/dashboard/DashboardPage";
import {shallow} from "enzyme"
import {fireEvent, screen} from "@testing-library/react";
import DashboardOrderPage from "@/components/screens/dashboard-order/DashboardOrderPage";
import OrdersCatalogItem from "@/components/screens/dashboard-order/orders-catalog-item/OrdersCatalogItem";
import {ORDERS_TEST, SAILS_TEST} from "@/components/screens/dashboard-order/orders-catalog/__tests__/constants";
import {useDeleteSailMutation} from "@/components/screens/dashboard-order/sail-item/useDeleteSailMutation.hook";
import {useUpdateSailMutation} from "@/components/screens/dashboard-order/sail-item/useUpdateSailMutation.hook";
import {useUpdateOrderMutationHook} from "@/components/screens/dashboard-order/orders-catalog-item/useUpdateOrderMutation.hook";
import {
    useCreateSailMutation
} from "@/components/screens/dashboard-order/orders-catalog-item/useCreateSailMutation.hook";
import SailItem from "@/components/screens/dashboard-order/sail-item/SailItem";
import {act} from "react-dom/test-utils";

jest.mock('next/navigation', () => ({
    useRouter: () => ({
        replace: jest.fn()
    })
}))

const mockedUpdateOrderMutate = jest.fn()
jest.mock("../useUpdateOrderMutation.hook")
jest.spyOn({useUpdateOrderMutation: useUpdateOrderMutationHook}, "useUpdateOrderMutation").mockImplementation(() => {
    return mockedUpdateOrderMutate
})

const mockedCreateSailMutate = jest.fn()
jest.mock("../useCreateSailMutation.hook")
jest.spyOn({useCreateSailMutation}, "useCreateSailMutation").mockImplementation(() => {
    return mockedCreateSailMutate
})
describe("Order catalog item", () => {

    afterEach(() => {
        jest.clearAllMocks()
    })

    test("Order catalog item render", () => {
        const component = renderWithProviders(<OrdersCatalogItem order={ORDERS_TEST[0]} />)

        expect(screen.getByText(ORDERS_TEST[0].total))
        expect(component).toMatchSnapshot()
    })

    test("Should call useCreateSailMutation with click", () => {
        const component = renderWithProviders(<OrdersCatalogItem order={ORDERS_TEST[0]} />)
        fireEvent.click(component.getByTestId("button-delete-test"))
        expect(mockedDeleteMutate).toHaveBeenCalledTimes(1)
    })

    test("Should call useUpdateSailMutation with click", () => {
        const component = renderWithProviders(<SailItem sail={SAILS_TEST[0]} />)
        act(() => {
            fireEvent.click(component.getByTestId('button-edit-test'))
        })

        fireEvent.click(component.getByTestId("button-confirm-test"))
        expect(mockedUpdateMutate).toHaveBeenCalledTimes(1)
    })

})