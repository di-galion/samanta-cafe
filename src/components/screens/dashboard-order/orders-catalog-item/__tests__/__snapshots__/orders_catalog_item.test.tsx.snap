// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order catalog item Order catalog item render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="item"
      >
        <form
          data-testid="order-item-test"
        >
          <div
            class="options_wrapper"
          >
            <div
              class="select"
            >
              <div
                class="select_wrapper"
                data-testod="change-status-test"
              >
                <button
                  class="select"
                  data-testid="select-button-test"
                  type="button"
                >
                  не подтверждено
                </button>
                <ul
                  class="list_hidden"
                  data-testid="select-list-test"
                >
                  <li
                    class="item"
                    data-value="не подтверждено"
                  >
                    не подтверждено
                  </li>
                  <li
                    class="item"
                    data-value="готовится"
                  >
                    готовится
                  </li>
                  <li
                    class="item"
                    data-value="ожидает клиента"
                  >
                    ожидает клиента
                  </li>
                  <li
                    class="item"
                    data-value="в доставке"
                  >
                    в доставке
                  </li>
                  <li
                    class="item"
                    data-value="отменен"
                  >
                    отменен
                  </li>
                </ul>
                <div
                  class="overflow"
                />
              </div>
            </div>
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
              />
            </svg>
          </div>
          <div
            class="inputs_section"
          >
            <div
              class="row"
            >
              <p>
                Имя
              </p>
              <input
                class="input"
                disabled=""
                id="Имя"
                name="Имя"
                value="Dima"
              />
            </div>
            <div
              class="row"
            >
              <p>
                Телефон
              </p>
              <input
                class="input"
                disabled=""
                id="Телефон"
                name="Телефон"
                value="89148508355"
              />
            </div>
            <div
              class="row"
            >
              <p>
                Улица
              </p>
              <input
                class="input"
                disabled=""
                id="Улица"
                name="Улица"
                value="Garlanovo"
              />
            </div>
            <div
              class="row"
            >
              <p>
                Дом
              </p>
              <input
                class="input"
                disabled=""
                id="Дом"
                name="Дом"
                value="234"
              />
            </div>
            <div
              class="row"
            >
              <p>
                Подьезд
              </p>
              <input
                class="input"
                disabled=""
                id="Подьезд"
                name="Подьезд"
                value="2"
              />
            </div>
            <div
              class="row"
            >
              <p>
                Этаж
              </p>
              <input
                class="input"
                disabled=""
                id="Этаж"
                name="Этаж"
                value="12"
              />
            </div>
            <div
              class="row"
            >
              <p>
                Комната
              </p>
              <input
                class="input"
                disabled=""
                id="Комната"
                name="Комната"
                value="154"
              />
            </div>
            <div
              class="row"
            >
              <p>
                Ключ
              </p>
              <input
                class="input"
                disabled=""
                id="Ключ"
                name="Ключ"
                value="#1245"
              />
            </div>
            <div
              class="row"
            >
              <p>
                Комментарий
              </p>
              <input
                class="input"
                disabled=""
                id="Комментарий"
                name="Комментарий"
                value="Some comment"
              />
            </div>
            <div
              class="children"
            />
          </div>
          <div
            class="sails"
          >
            <div
              class="sail"
            >
              <form
                class="info_section"
                data-testid="sail-item-test"
                id="form-sail 1"
              >
                <div
                  class="options_wrapper"
                >
                  <svg
                    data-testid="button-edit-test"
                    fill="currentColor"
                    height="23"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="23"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                    />
                  </svg>
                  <svg
                    class="delete"
                    data-testid="button-delete-test"
                    fill="none"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 15 15"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12.8536 2.85355C13.0488 2.65829 13.0488 2.34171 12.8536 2.14645C12.6583 1.95118 12.3417 1.95118 12.1464 2.14645L7.5 6.79289L2.85355 2.14645C2.65829 1.95118 2.34171 1.95118 2.14645 2.14645C1.95118 2.34171 1.95118 2.65829 2.14645 2.85355L6.79289 7.5L2.14645 12.1464C1.95118 12.3417 1.95118 12.6583 2.14645 12.8536C2.34171 13.0488 2.65829 13.0488 2.85355 12.8536L7.5 8.20711L12.1464 12.8536C12.3417 13.0488 12.6583 13.0488 12.8536 12.8536C13.0488 12.6583 13.0488 12.3417 12.8536 12.1464L8.20711 7.5L12.8536 2.85355Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
                <div
                  class="input_wrapper"
                >
                  <label
                    for="Sail"
                  >
                    Name:
                  </label>
                  <input
                    class="input"
                    disabled=""
                    name="Sail"
                    value="Sail"
                  />
                  <label
                    for="124"
                  >
                    Price:
                  </label>
                  <input
                    class="input"
                    disabled=""
                    name="124"
                    value="124"
                  />
                  <label
                    for="1"
                  >
                    Quantity:
                  </label>
                  <input
                    class="input"
                    disabled=""
                    name="1"
                    value="1"
                  />
                </div>
              </form>
            </div>
          </div>
          <div
            class="button_add_sails"
          >
            <button
              class="main"
              data-testid="button-add-sails-test"
            >
              Add sails
            </button>
          </div>
          <div
            class="total"
          >
            1245
          </div>
        </form>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="item"
    >
      <form
        data-testid="order-item-test"
      >
        <div
          class="options_wrapper"
        >
          <div
            class="select"
          >
            <div
              class="select_wrapper"
              data-testod="change-status-test"
            >
              <button
                class="select"
                data-testid="select-button-test"
                type="button"
              >
                не подтверждено
              </button>
              <ul
                class="list_hidden"
                data-testid="select-list-test"
              >
                <li
                  class="item"
                  data-value="не подтверждено"
                >
                  не подтверждено
                </li>
                <li
                  class="item"
                  data-value="готовится"
                >
                  готовится
                </li>
                <li
                  class="item"
                  data-value="ожидает клиента"
                >
                  ожидает клиента
                </li>
                <li
                  class="item"
                  data-value="в доставке"
                >
                  в доставке
                </li>
                <li
                  class="item"
                  data-value="отменен"
                >
                  отменен
                </li>
              </ul>
              <div
                class="overflow"
              />
            </div>
          </div>
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
            />
          </svg>
        </div>
        <div
          class="inputs_section"
        >
          <div
            class="row"
          >
            <p>
              Имя
            </p>
            <input
              class="input"
              disabled=""
              id="Имя"
              name="Имя"
              value="Dima"
            />
          </div>
          <div
            class="row"
          >
            <p>
              Телефон
            </p>
            <input
              class="input"
              disabled=""
              id="Телефон"
              name="Телефон"
              value="89148508355"
            />
          </div>
          <div
            class="row"
          >
            <p>
              Улица
            </p>
            <input
              class="input"
              disabled=""
              id="Улица"
              name="Улица"
              value="Garlanovo"
            />
          </div>
          <div
            class="row"
          >
            <p>
              Дом
            </p>
            <input
              class="input"
              disabled=""
              id="Дом"
              name="Дом"
              value="234"
            />
          </div>
          <div
            class="row"
          >
            <p>
              Подьезд
            </p>
            <input
              class="input"
              disabled=""
              id="Подьезд"
              name="Подьезд"
              value="2"
            />
          </div>
          <div
            class="row"
          >
            <p>
              Этаж
            </p>
            <input
              class="input"
              disabled=""
              id="Этаж"
              name="Этаж"
              value="12"
            />
          </div>
          <div
            class="row"
          >
            <p>
              Комната
            </p>
            <input
              class="input"
              disabled=""
              id="Комната"
              name="Комната"
              value="154"
            />
          </div>
          <div
            class="row"
          >
            <p>
              Ключ
            </p>
            <input
              class="input"
              disabled=""
              id="Ключ"
              name="Ключ"
              value="#1245"
            />
          </div>
          <div
            class="row"
          >
            <p>
              Комментарий
            </p>
            <input
              class="input"
              disabled=""
              id="Комментарий"
              name="Комментарий"
              value="Some comment"
            />
          </div>
          <div
            class="children"
          />
        </div>
        <div
          class="sails"
        >
          <div
            class="sail"
          >
            <form
              class="info_section"
              data-testid="sail-item-test"
              id="form-sail 1"
            >
              <div
                class="options_wrapper"
              >
                <svg
                  data-testid="button-edit-test"
                  fill="currentColor"
                  height="23"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="23"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19.045 7.401c.378-.378.586-.88.586-1.414s-.208-1.036-.586-1.414l-1.586-1.586c-.378-.378-.88-.586-1.414-.586s-1.036.208-1.413.585L4 13.585V18h4.413L19.045 7.401zm-3-3 1.587 1.585-1.59 1.584-1.586-1.585 1.589-1.584zM6 16v-1.585l7.04-7.018 1.586 1.586L7.587 16H6zm-2 4h16v2H4z"
                  />
                </svg>
                <svg
                  class="delete"
                  data-testid="button-delete-test"
                  fill="none"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 15 15"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12.8536 2.85355C13.0488 2.65829 13.0488 2.34171 12.8536 2.14645C12.6583 1.95118 12.3417 1.95118 12.1464 2.14645L7.5 6.79289L2.85355 2.14645C2.65829 1.95118 2.34171 1.95118 2.14645 2.14645C1.95118 2.34171 1.95118 2.65829 2.14645 2.85355L6.79289 7.5L2.14645 12.1464C1.95118 12.3417 1.95118 12.6583 2.14645 12.8536C2.34171 13.0488 2.65829 13.0488 2.85355 12.8536L7.5 8.20711L12.1464 12.8536C12.3417 13.0488 12.6583 13.0488 12.8536 12.8536C13.0488 12.6583 13.0488 12.3417 12.8536 12.1464L8.20711 7.5L12.8536 2.85355Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <div
                class="input_wrapper"
              >
                <label
                  for="Sail"
                >
                  Name:
                </label>
                <input
                  class="input"
                  disabled=""
                  name="Sail"
                  value="Sail"
                />
                <label
                  for="124"
                >
                  Price:
                </label>
                <input
                  class="input"
                  disabled=""
                  name="124"
                  value="124"
                />
                <label
                  for="1"
                >
                  Quantity:
                </label>
                <input
                  class="input"
                  disabled=""
                  name="1"
                  value="1"
                />
              </div>
            </form>
          </div>
        </div>
        <div
          class="button_add_sails"
        >
          <button
            class="main"
            data-testid="button-add-sails-test"
          >
            Add sails
          </button>
        </div>
        <div
          class="total"
        >
          1245
        </div>
      </form>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
